openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py

```diff
@@ -0,0 +1,150 @@
+                json=data,
+            )
+        except Exception as error:
+            msg_error = 'Failed to authenticate at MIT HZ API. Reason: %s'
+            log_details['response'] = {'error': msg_error + str(error)}
+            LOG.error(msg_error + str(error))
+            EnrollmentRequestLog.objects.create(
+                request_type=str(self),
+                details=log_details,
+            )
+        else:
+            if not response.ok:
+                log_details['response'] = response.text()
+                EnrollmentRequestLog.objects.create(
+                    request_type=str(self),
+                    details=log_details,
+                )
+                LOG.error('failed when trying to authenticate with MIT HORIZON')
+                return 'missing token'
+            return response.json()['access_token']
+
+    def _get_enrollment_headers(self):
+        """
+        Formats headers for MIT_HZ API.
+        """
+        return {
+            'Authorization': 'Bearer {token}'.format(token=self._get_bearer_token()),
+            'Content-Type': 'application/json',
+        }
+
+    def _check_user(self, user_id):
+        """
+        Check if user exists in MIT HORIZON.
+        """
+        headers = self._get_enrollment_headers()
+        url = '{root_url}{get_user_path}{user_id}'.format(
+            root_url=settings.MIT_HZ_API_URL,
+            get_user_path=settings.MIT_HZ_GET_USER_PATH,
+            user_id=user_id
+        )
+        log_details = {
+            'url': url,
+            'user': user_id,
+        }
+
+        try:
+            response = requests.get(url, headers=headers)
+        except Exception as error:
+            LOG.error('Failed to get user at MIT HZ API.')
+            log_details['response'] = {'error': 'Failed to get user at MIT HZ API. Reason: %s' + str(error)}
+            EnrollmentRequestLog.objects.create(
+                request_type=str(self),
+                details=log_details,
+            )
+            return str(error), status.HTTP_500_INTERNAL_SERVER_ERROR
+        else:
+            if not response.ok:
+                log_details['response'] = response.text()
+                LOG.error('User not found at MIT HZ API.')
+                EnrollmentRequestLog.objects.create(
+                    request_type=str(self),
+                    details=log_details,
+                )
+                return False
+
+            log_details['response'] = response.json()
+            log_details['message'] = 'User found'
+            EnrollmentRequestLog.objects.create(
+                request_type=str(self),
+                details=log_details,
+            )
+            return True
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        Returns the data required to refresh a user in the MIT HORIZON API.
+        """
+        enrollment_data = {}
+        user_id = self._get_user_id(data.get('user_email'))
```

Comment:
Does '_get_user_id' email param nullable? @luismorenolopera.

Reply:
I don't think this case is possible @ivanvgh